### PR TITLE
Activity Log: Introduce event sharing

### PR DESCRIPTION
**Pull request by rcoll**
#### Changes proposed in this Pull Request

* Introduces event sharing on the activity log in Calypso Green.
* Add state and data layer support for POSTing to `/activity-log/{siteId}/share/{activityId}`.
* Adds rough css to style the sharing button and popover.
* Adds success and error notices to indicate whether sharing has succeeded or failed.

#### Testing instructions

* Visit the activity log in Calypso Green dev. You should see "Share this event" at the bottom of most events.
* Clicking the share button should display a popover with an email address field and submit button.
* Email addresses should loosely validate.
* Submitting the form should result in an error notice (the API is not yet shipped).

Related to #

### CODE REVIEW

**Comment by keoshi** [comment]
Hey @rcoll — hope you don't mind but I jumped in and made some tweaks. Most notably:

- Change the placement of the share button because of the conflict with the actions button you brought up in Slack.
- In regards to streams, the share button is only visible in parent events now.
- Inside the form, the Share button is now disabled unless the input has some data in it.
- Added corresponding styles for these changes, increased the popover width a tiny bit, and polished up some spacings.

The good thing about this new location is that mobile devices are taken care of fully and the behavior stays consistent across the board. Here's a quick overview of how it looks right now:

https://user-images.githubusercontent.com/390760/121337233-3a535200-c914-11eb-831e-c364e92f1007.mp4

File: client/state/data-layer/wpcom/activity-log/share/index.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+		{
+			apiVersion: '1',
+			method: 'POST',
+			path: `/activity-log/${ action.siteId }/share/${ action.activityId }`,
+			body: {
+				email: action.email,
+			},
+		},
+		action
+	);
```
Comment (atanas-dev) [inline client/state/data-layer/wpcom/activity-log/share/index.js]: Here's what an example API request would look like this:
`/wpcom/v2/sites/{BLOG_ID}/activity/{REWIND_ID}/share?email={RECIPIENT_EMAIL}`
Note that it uses a rewind id and not an activity id due to Rewind internals (each activity should have a rewind_id prop).

Comment (rcoll) [reply client/state/data-layer/wpcom/activity-log/share/index.js]: Updated PR to reflect proper API path. Please note that the email field will be urlencoded.

Comment (rcoll) [reply client/state/data-layer/wpcom/activity-log/share/index.js]: Also updated to use rewindId :-)

**Review by atanas-dev — APPROVED** [review]
LGTM :+1:
I personally would have split the "Share this event" button into its own component but this is obviously non-blocking.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6026542

Hi @rcoll, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Share this event`
- `Share this event via email`
- `Share what is happening with your site with your clients or business partners.`
- `We've shared the event!`
- `The event failed to send, please try again.`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6026542).

**Pull request merged**